controllers/drplacementcontrol.go

```diff
@@ -1513,6 +1513,19 @@ func (d *DRPCInstance) generateVRG(repState rmn.ReplicationState) rmn.VolumeRepl
 	return vrg
 }
 
+func (d *DRPCInstance) availableS3Profiles() []string {
+	r := []string{}
+
+	for i := range d.drClusters {
+		c := &d.drClusters[i]
+		if !drClusterIsDeleted(c) {
+			r = append(r, c.Spec.S3ProfileName)
```

Comment:
This does not work - we see in the logs that the vrg is updated using both clusters, so the drcluster object we use here must be stale.


1. drcluster controller find tha the drcluster was deleted

```
2023-12-12T16:15:34.356Z	INFO	controllers.DRCluster	controllers/drcluster_controller.go:272	reconcile enter	{"name": "dr1", "rid": "bf896718-4c69-41a0-bb95-9182e04a42ac"}
2023-12-12T16:15:34.356Z	INFO	controllers.DRCluster	controllers/drcluster_controller.go:453	delete	{"name": "dr1", "rid": "bf896718-4c69-41a0-bb95-9182e04a42ac"}
2023-12-12T16:15:34.357Z	INFO	controllers.DRCluster	controllers/drcluster_controller.go:298	reconcile exit	{"name": "dr1", "rid": "bf896718-4c69-41a0-bb95-9182e04a42ac"}
2023-12-12T16:15:34.357Z	ERROR	controller/controller.go:329	Reconciler error	{"controller": "drcluster", "controllerGroup": "ramendr.openshift.io", "controllerKind": "DRCluster", "DRCluster": {"name":"dr1"}, "namespace": "", "name": "dr1", "reconcileID": "33b6081b-3d31-48a0-866c-88ab6a038e7b", "error": "drclusters undeploy: drcluster 'dr1' referenced in one or more existing drPolicy resources"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:329
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:274
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:235
```

2. 7 minutes later - drplacement controller detects that the cluster was deleted

```
2023-12-12T16:23:29.231Z	INFO	controllers.DRPlacementControl	controllers/drplacementcontrol_controller.go:594	Entering reconcile loop	{"DRPC": "busybox-regional-rbd-deploy/busybox-regional-rbd-deploy-drpc", "rid": "2e059f82-a46a-48bd-b975-a0619a608aa7"}
...
2023-12-12T16:23:29.234Z	INFO	controllers.DRPlacementControl	controllers/drplacementcontrol_controller.go:1486	Skipping VRG on deleted drcluster	{"DRPC": "busybox-regional-rbd-deploy/busybox-regional-rbd-deploy-drpc", "rid": "2e059f82-a46a-48bd-b975-a0619a608aa7", "drcluster": "dr1", "vrg": "busybox-regional-rbd-deploy-drpc"}
```

3. In the same reconcile loop - it updates the vrg manifestwork - including s3 store on the deleted cluster

```
2023-12-12T16:23:29.236Z	INFO	controllers.DRPlacementControl	controllers/drplacementcontrol.go:1606	ensureNamespaceExistsOnManagedCluster: namespace 'busybox-regional-rbd-deploy' exists on cluster dr2: true	{"DRPC": "busybox-regional-rbd-deploy/busybox-regional-rbd-deploy-drpc", "rid": "2e059f82-a46a-48bd-b975-a0619a608aa7"}
2023-12-12T16:23:29.236Z	INFO	controllers.DRPlacementControl	controllers/drplacementcontrol.go:1434	Creating VRG ManifestWork	{"DRPC": "busybox-regional-rbd-deploy/busybox-regional-rbd-deploy-drpc", "rid": "2e059f82-a46a-48bd-b975-a0619a608aa7", "Last State:": "FailedOver", "cluster": "dr2"}
2023-12-12T16:23:29.236Z	INFO	controllers.DRPlacementControl	util/mw_util.go:121	Create or Update manifestwork busybox-regional-rbd-deploy-drpc:busybox-regional-rbd-deploy:dr2:{TypeMeta:{Kind:VolumeReplicationGroup APIVersion:ramendr.openshift.io/v1alpha1} ObjectMeta:{Name:busybox-regional-rbd-deploy-drpc GenerateName: Namespace:busybox-regional-rbd-deploy SelfLink: UID: ResourceVersion: Generation:0 CreationTimestamp:0001-01-01 00:00:00 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[] Annotations:map[] OwnerReferences:[] Finalizers:[] ManagedFields:[]} Spec:{PVCSelector:{MatchLabels:map[appname:busybox] MatchExpressions:[]} ReplicationState:primary S3Profiles:[minio-on-dr2] Async:0xc000fe3a90 Sync:<nil> VolSync:{RDSpec:[] Disabled:false} PrepareForFinalSync:false RunFinalSync:false Action:Failover KubeObjectProtection:<nil>} Status:{State: ProtectedPVCs:[] Conditions:[] ObservedGeneration:0 LastUpdateTime:0001-01-01 00:00:00 +0000 UTC KubeObjectProtection:{CaptureToRecoverFrom:<nil>} PrepareForFinalSyncComplete:false FinalSyncComplete:false LastGroupSyncTime:<nil> LastGroupSyncDuration:nil LastGroupSyncBytes:<nil>}}	{"DRPC": "busybox-regional-rbd-deploy/busybox-regional-rbd-deploy-drpc", "rid": "2e059f82-a46a-48bd-b975-a0619a608aa7"}
2023-12-12T16:23:29.236Z	INFO	controllers.DRPlacementControl	util/mw_util.go:499	Updating Manifestwork	{"DRPC": "busybox-regional-rbd-deploy/busybox-regional-rbd-deploy-drpc", "rid": "2e059f82-a46a-48bd-b975-a0619a608aa7", "name": "busybox-regional-rbd-deploy-drpc-busybox-regional-rbd-deploy-vrg-mw", "namespace": "dr2"}
```